lib/active_model/serializer/adapter/json.rb

```diff
@@ -6,6 +6,8 @@ class Adapter
       class Json < Adapter
         def serializable_hash(options = nil)
           options ||= {}
+          sideload = ActiveModel::Serializer.config.sideload_associations
```

Comment:
not entirely related to the PR, but should this method be recursive? it looks like for the associations, the hash is being manually built.